### ISSUE TITLE
Regression: Unable to push windows container images in 1.1.1

### ISSUE THREAD

**Issue by nlowe** [vmware/harbor#2349]
As I mentioned in #2328, I needed to manually edit the `docker-compose.yml` file to make the registry use `vmware/registry:2.6.1-photon` instead of `vmware/registry:photon-2.6.0`, otherwise I could not push windows container images. I spun up a `0.5` version of harbor and was able to push windows container images without any configuration changes using the version of the registry that shipped with `0.5`.

### Failing Example

Here's what pushing a windows container looks like form the client. All layers get pushed, but it looks like when the client pushes the manifest it errors out:

```powershell
PS C:\Windows\system32> docker login harbor.mydomain.net
Username (admin): admin
Password:
Login Succeeded
PS C:\Windows\system32> docker push harbor.mydomain.net/test/servercore-base
The push refers to a repository [harbor.mydomain.net/test/servercore-base]
7d549ef3966a: Pushed
64195430ffc9: Pushed
ff20a3ca4594: Pushed
7ca4db46ccb0: Pushed
d48b65fa3949: Pushed
f05bb0c81ca2: Pushed
75a8871d89f0: Pushed
b03c435753e9: Pushed
bfbc6476f60b: Pushed
197c864e1b7a: Pushed
ce74595d58f5: Pushed
5b4aace84103: Pushed
1f2f3eb32edc: Pushed
0451551dda21: Pushed
c28d44287ce5: Skipped foreign layer
f358be10862c: Skipped foreign layer
errors:
manifest blob unknown: blob unknown to registry
manifest blob unknown: blob unknown to registry
```

Relevant logs from when the error was thrown:

```
harbor-ui      | 2017/05/19 16:21:30 [router.go:829][D] | POST       | /service/notifications                   | 1.982636ms       | match      | /service/notifications                   |
registry       | time="2017-05-19T16:21:30.955437765Z" level=debug msg="authorizing request" go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.956663636Z" level=debug msg=PutImageManifest auth.user.name=admin go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.957660683Z" level=debug msg="(*manifestStore).Put" auth.user.name=admin go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.957752815Z" level=debug msg="(*schema2ManifestHandler).Put" auth.user.name=admin go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.958938926Z" level=debug msg="filesystem.GetContent(\"/docker/registry/v2/repositories/test/servercore-base/_layers/sha256/3889bb8d808bbae6fa5a33e07093e65c31371bcf9e4c38c21be6b9af52ad1548/link\")" auth.user.name=admin go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry trace.duration=78.13µs trace.file="/root/gows/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).GetContent" trace.id=be2d85f9-3760-4726-baa3-fc59e4e45bce trace.line=82 vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.959756852Z" level=debug msg="filesystem.GetContent(\"/docker/registry/v2/repositories/test/servercore-base/_layers/sha256/3430754e4d171ead00cf6766797a28abf3caf236f6c92c5c346ea2ad3955a129/link\")" auth.user.name=admin go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry trace.duration=34.693µs trace.file="/root/gows/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).GetContent" trace.id=38373454-194c-4ed7-9c19-255b06e7052a trace.line=82 vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.960638567Z" level=error msg="response completed with error" auth.user.name=admin err.code="manifest blob unknown" err.detail=sha256:3889bb8d808bbae6fa5a33e07093e65c31371bcf9e4c38c21be6b9af52ad1548 err.message="blob unknown to registry" go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" http.response.contenttype="application/json; charset=utf-8" http.response.duration=6.818714ms http.response.status=400 http.response.written=319 instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | time="2017-05-19T16:21:30.960781913Z" level=error msg="response completed with error" auth.user.name=admin err.code="manifest blob unknown" err.detail=sha256:3430754e4d171ead00cf6766797a28abf3caf236f6c92c5c346ea2ad3955a129 err.message="blob unknown to registry" go.version=go1.7.3 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=3f127232-3e43-4c1d-a8a3-593df0f5d427 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" http.response.contenttype="application/json; charset=utf-8" http.response.duration=6.986353ms http.response.status=400 http.response.written=319 instance.id=235246ca-5266-431d-aacb-805973457ef9 service=registry vars.name="test/servercore-base" vars.reference=latest version="v2.4.1+unknown"
registry       | 172.19.0.8 - - [19/May/2017:16:21:30 +0000] "PUT /v2/test/servercore-base/manifests/latest HTTP/1.1" 400 319 "" "docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))"
nginx          | 10.40.3.46 - "PUT /v2/test/servercore-base/manifests/latest HTTP/1.1" 400 319 "-" "docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \x5C(windows\x5C))" 0.009 0.009 .
```

### Working Example

Here's what pushing looks like on `1.1.1` using the latest registry image:

```powershell
PS C:\Windows\system32> docker login harbor.mydomain.net
Username (admin): admin
Password:
Login Succeeded
PS C:\Windows\system32> docker push harbor.mydomain.net/test/servercore-base
The push refers to a repository [harbor.mydomain.net/test/servercore-base]
7d549ef3966a: Pushed
64195430ffc9: Pushed
ff20a3ca4594: Pushed
7ca4db46ccb0: Pushed
d48b65fa3949: Pushed
f05bb0c81ca2: Pushed
75a8871d89f0: Pushed
b03c435753e9: Pushed
bfbc6476f60b: Pushed
197c864e1b7a: Pushed
ce74595d58f5: Pushed
5b4aace84103: Pushed
1f2f3eb32edc: Pushed
0451551dda21: Pushed
c28d44287ce5: Skipped foreign layer
f358be10862c: Skipped foreign layer
latest: digest: sha256:f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399 size: 3890
```

And logs from creating the manifest:

```
registry       | time="2017-05-19T16:30:51.117148187Z" level=debug msg="authorizing request" go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.118821311Z" level=debug msg=PutImageManifest auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.129233458Z" level=debug msg="(*manifestStore).Put" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.129515724Z" level=debug msg="(*schema2ManifestHandler).Put" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.131344346Z" level=debug msg="filesystem.Stat(\"/docker/registry/v2/blobs/sha256/f1/f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399/data\")" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry trace.duration=60.265µs trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).Stat" trace.id=699da5e5-80bf-4e29-bb1d-25f9b6aa3845 trace.line=137 vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.141055267Z" level=debug msg="filesystem.PutContent(\"/docker/registry/v2/blobs/sha256/f1/f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399/data\")" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry trace.duration=9.434187ms trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).PutContent" trace.id=9deed7a2-d5c5-46ab-b61d-c118f50a6eee trace.line=95 vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.144610173Z" level=debug msg="filesystem.PutContent(\"/docker/registry/v2/repositories/test/servercore-base/_manifests/revisions/sha256/f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399/link\")" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry trace.duration=2.634415ms trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).PutContent" trace.id=8560ebb2-ab40-48cc-b13b-ef9b77bb2ea0 trace.line=95 vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.149588771Z" level=debug msg="filesystem.PutContent(\"/docker/registry/v2/repositories/test/servercore-base/_manifests/tags/latest/index/sha256/f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399/link\")" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry trace.duration=2.44333ms trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).PutContent" trace.id=518f9b48-51e5-460f-b58e-5853014c1a60 trace.line=95 vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.152535758Z" level=debug msg="filesystem.PutContent(\"/docker/registry/v2/repositories/test/servercore-base/_manifests/tags/latest/current/link\")" auth.user.name=admin go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry trace.duration=2.783697ms trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).PutContent" trace.id=d97f2918-3f64-4dc7-b40b-1fb8add278df trace.line=95 vars.name="test/servercore-base" vars.reference=latest version=v2.6.1
registry       | time="2017-05-19T16:30:51.153458398Z" level=info msg="response completed" go.version=go1.7.5 http.request.contenttype="application/vnd.docker.distribution.manifest.v2+json" http.request.host=harbor.mydomain.net http.request.id=ff824667-9bcd-49cc-9872-b25dcf6ac939 http.request.method=PUT http.request.remoteaddr=10.40.3.46 http.request.uri="/v2/test/servercore-base/manifests/latest" http.request.useragent="docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))" http.response.duration=38.214409ms http.response.status=201 http.response.written=0 instance.id=189512f1-4507-488e-82b8-f8303f22bb5c service=registry version=v2.6.1
registry       | 172.19.0.8 - - [19/May/2017:16:30:51 +0000] "PUT /v2/test/servercore-base/manifests/latest HTTP/1.1" 201 0 "" "docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \\(windows\\))"
nginx          | 10.40.3.46 - "PUT /v2/test/servercore-base/manifests/latest HTTP/1.1" 201 0 "-" "docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \x5C(windows\x5C))" 0.040 0.040 .
harbor-ui      | 2017-05-19T16:30:51Z [DEBUG] [notification.go:102]: receive an event:
harbor-ui      | ----ID: 8d4f3105-8e96-4319-8f3f-852d7acd9f97
harbor-ui      | ----target: test/servercore-base:latest
harbor-ui      | ----digest: sha256:f17322546599f9b8f18e2617cae47a9234bcbd1d2ba91f2a604140337bb67399
harbor-ui      | ----action: push
harbor-ui      | ----mediatype: application/vnd.docker.distribution.manifest.v2+json
harbor-ui      | ----user-agent: docker/1.12.2-cs2-ws-beta go/go1.7.1 git-commit/050b611 os/windows arch/amd64 UpstreamClient(Docker-Client/1.12.2-cs2-ws-beta \(windows\))
harbor-ui      | 2017-05-19T16:30:51Z [DEBUG] [notification.go:118]: add event to collect: 8d4f3105-8e96-4319-8f3f-852d7acd9f97
harbor-ui      | 2017/05/19 16:30:51 [router.go:829][D] | POST       | /service/notifications                   | 1.838351ms       | match      | /service/notifications                   |
harbor-ui      | 2017-05-19T16:30:51Z [DEBUG] [notification.go:78]: Add repository test/servercore-base into DB.
```

#### Client
docker:

```powershell
PS C:\Windows\system32> docker info
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 11
Server Version: 1.12.2-cs2-ws-beta
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: nat null overlay
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.693.amd64fre.rs1_release.161220-1747)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 3.999 GiB
Name: QAV-003080
ID: KB54:NWOV:2PHV:I2LT:LFQI:GIOR:SWC5:4ZIZ:D4PS:3FJP:RVUA:JGBW
Docker Root Dir: M:\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

docker-compose:

```
PS C:\Windows\system32> docker-compose --version
docker-compose version 1.11.1, build 7c5d5e4
```

#### Server
docker:

```bash
user@harbor:~$ docker info
Containers: 7
 Running: 7
 Paused: 0
 Stopped: 0
Images: 25
Server Version: 1.12.6
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 124
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-78-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.953 GiB
Name: harbor
ID: TYX3:PW2V:MSQQ:77EE:BRV5:M5BA:DACM:6NFJ:KXS5:PIDV:6Z2X:HNGP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

docker-compose:

```bash
user@harbor:~$ docker-compose --version
docker-compose version 1.8.0, build unknown
```

### harbor config
I used the default harbor config for these tests.

## Workaround
I was able to push windows container images by editing `docker-compose.yml` to use the latest tag for the registry:

```yml
#...
  registry:
    image: vmware/registry:2.6.1-photon
    container_name: registry
    restart: always
    volumes:
      - /data/registry:/storage:z
      - ./common/config/registry/:/etc/registry/:z
    networks:
      - harbor
    environment:
      - GODEBUG=netdns=cgo
    command:
      ["serve", "/etc/registry/config.yml"]
    depends_on:
      - log
    logging:
      driver: "syslog"
      options:
        syslog-address: "tcp://127.0.0.1:1514"
        tag: "registry"
# ...
```

**Comment by ywk253100** [comment]
It should be fixed in branch 1.1.0 and new release will be available soon.

**Comment by reasonerjt** [comment]
This is fixed in #2328, I'll keep this open when 1.1.2 is released.
